crates/solver/src/settlement_submission/submitter/common.rs

```diff
@@ -1,88 +1,62 @@
-    let body = response
-        .text()
-        .await
-        .map_err(|err| SubmitApiError::Other(err.into()))?;
+/// An additonal specialized submitter API for private network transactions.
+#[derive(Clone)]
+pub struct PrivateNetwork(Web3);
 
-    let handle = parse_json_rpc_response::<H256>(&body)?;
-    tracing::info!(
-        "created transaction with hash: {:?} and handle: {:?}, url: {}",
-        tx_hash,
-        handle,
-        url.as_str()
-    );
-    Ok(TransactionHandle { tx_hash, handle })
+impl Namespace<Web3Transport> for PrivateNetwork {
+    fn new(transport: Web3Transport) -> Self {
+        Self(Web3::new(transport))
+    }
+
+    fn transport(&self) -> &Web3Transport {
+        self.0.transport()
+    }
 }
 
-fn parse_json_rpc_response<T>(body: &str) -> Result<T, SubmitApiError>
-where
-    T: DeserializeOwned,
-{
-    match serde_json::from_str::<Output>(body) {
-        Ok(output) => match output {
-            Output::Success(body) => serde_json::from_value::<T>(body.result).map_err(|_| {
-                anyhow!(
-                    "failed conversion to expected type {}",
-                    std::any::type_name::<T>()
-                )
-                .into()
-            }),
-            Output::Failure(body) => {
-                if body.error.message.starts_with("invalid nonce")
-                    || body.error.message.starts_with("nonce too low")
-                {
-                    Err(SubmitApiError::InvalidNonce)
-                } else if body
-                    .error
-                    .message
-                    .starts_with("Transaction gas price supplied is too low")
-                {
-                    Err(SubmitApiError::OpenEthereumTooCheapToReplace)
-                } else if body
-                    .error
-                    .message
-                    .starts_with("replacement transaction underpriced")
-                {
-                    Err(SubmitApiError::ReplacementTransactionUnderpriced)
-                } else {
-                    Err(anyhow!("rpc error: {}", body.error).into())
-                }
-            }
-        },
-        Err(_) => Err(anyhow!("invalid rpc response: {}", body).into()),
+impl PrivateNetwork {
+    /// Function for sending raw signed transaction to private networks
+    pub async fn submit_raw_transaction(
+        &self,
+        tx: TransactionBuilder<Web3Transport>,
+    ) -> Result<TransactionHandle, SubmitApiError> {
+        let (raw_signed_transaction, tx_hash) = match tx.build().now_or_never().unwrap().unwrap() {
+            Transaction::Request(_) => unreachable!("verified offline account was used"),
+            Transaction::Raw { bytes, hash } => (bytes.0, hash),
+        };
+
+        let handle = self
+            .0
+            .eth()
+            .send_raw_transaction(Bytes(raw_signed_transaction))
+            .await
+            .map_err(convert_web3_to_submission_error)?;
+
```

Comment:
Here I would revert the deleted log:
`tracing::info!(
        "created transaction with hash: {:?} and handle: {:?}, url: {}",
        tx_hash,
        handle,
        url.as_str()
    );
`

Reply:
Not sure if we have a type of private network now... (previously it was recognized by `url`)

Reply:
Right, didn't mean to remove it.

> Not sure if we have a type of private network now...

I added names to the HTTP transports, so you should see `[eden]` or `[flashbots]` in the JSON RPC logs around the message. We can also log it further up the call chain to have more context for the message.

Reply:
I added the log back in. Good catch!

https://github.com/gnosis/gp-v2-services/blob/65ab1cc090290023318ab7bb9520305dc26005b2/crates/solver/src/settlement_submission/submitter.rs#L441-L444

Note that I moved it higher up in the call-chain so that it can have more context (specifically, it now includes the submitter name instead of the URL).